src/lib/queryVersions.ts

```diff
@@ -37,79 +37,68 @@ async function queryVersions(packageMap: Index<VersionSpec>, options: Options =
   async function getPackageVersionProtected(dep: VersionSpec): Promise<VersionResult> {
     const npmAlias = parseNpmAlias(packageMap[dep])
     const [name, version] = npmAlias || [dep, packageMap[dep]]
-    const targetResult = typeof target === 'string' ? target : target(name, parseRange(version))
+    let targetResult = typeof target === 'string' ? target : target(name, parseRange(version))
+    let distTag = 'latest'
+
+    if (targetResult[0] === '@') {
+      distTag = targetResult.slice(1)
+      targetResult = 'distTag'
+    }
 
     let versionNew: Version | null = null
+    const isGithubDependency = isGithubUrl(packageMap[dep])
 
-    // use gitTags package manager for git urls
-    if (isGithubUrl(packageMap[dep])) {
-      // override packageManager and getPackageVersion just for this dependency
-      const packageManager = packageManagers.gitTags
-      const getPackageVersion = packageManager[targetResult as keyof typeof packageManager] as GetVersion
-
-      if (!getPackageVersion) {
-        const packageManagerSupportedVersionTargets = supportedVersionTargets.filter(t => t in packageManager)
-        return Promise.reject(
-          new Error(
-            `Unsupported target "${targetResult}" for github urls. Supported version targets are: ${packageManagerSupportedVersionTargets.join(
-              ', ',
-            )}`,
-          ),
-        )
-      }
+    // use gitTags package manager for git urls (for this dependency only)
+    const packageManager = isGithubDependency ? packageManagers.gitTags : globalPackageManager
+    const packageManagerName = isGithubDependency ? 'github urls' : options.packageManager || 'npm'
+
+    const getPackageVersion = packageManager[targetResult as keyof typeof packageManager] as GetVersion
+
+    if (!getPackageVersion) {
+      const packageManagerSupportedVersionTargets = supportedVersionTargets.filter(t => t in packageManager)
+      return Promise.reject(
+        new Error(
+          `Unsupported target "${targetResult}" for ${packageManagerName}. Supported version targets are: ` +
+            packageManagerSupportedVersionTargets.join(', ') +
+            (!isGithubDependency ? ' and custom distribution tags, following "@" (example: @next)' : ''),
+        ),
+      )
+    }
+
+    try {
```

Comment:
So much copy-paste :)

Couldn't take it, had to "insert" some features into the 'git url' condition branch. They were exclusive to npm before unification.

Amongst them:
- retry
- try/catch